### ISSUE TITLE
Make BMDB text shrinking logic smarter

### ISSUE THREAD

**Issue by arsalansufi** [votingworks/vxsuite#5514]
Rather than decreasing font size and activating "degrading a11y experience" mode based on the number contests, we could be smarter and try rendering on one page and, if we fail, shrink font size, maybe repeating a couple of times.

In our testing, we expectedly found that it's not just the number of contests that matters but contest title/option length, number of selections, etc. The only way to really know whether we need to shrink down is to try to render.

See https://votingworks.slack.com/archives/CJU9MSC6S/p1728593684882159?thread_ts=1728593684.882159&cid=CJU9MSC6S
